src/Http/Event/UpdateSubEventsRequestHandler.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Event;
+
+use Broadway\CommandHandling\CommandBus;
+use CultuurNet\UDB3\Event\Commands\UpdateSubEvents;
+use CultuurNet\UDB3\Event\ValueObjects\Status;
+use CultuurNet\UDB3\Event\ValueObjects\StatusReason;
+use CultuurNet\UDB3\Event\ValueObjects\StatusType;
+use CultuurNet\UDB3\Event\ValueObjects\SubEventUpdate;
+use CultuurNet\UDB3\Http\Response\NoContentResponse;
+use CultuurNet\UDB3\Language;
+use CultuurNet\UDB3\Offer\ValueObjects\BookingAvailability;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class UpdateSubEventsRequestHandler
+{
+    private CommandBus $commandBus;
+
+    private UpdateSubEventsRequestBodyParser $updateSubEventsParser;
+
+    public function __construct(
+        CommandBus $commandBus,
+        UpdateSubEventsRequestBodyParser $updateSubEventsParser
+    ) {
+        $this->commandBus = $commandBus;
+        $this->updateSubEventsParser = $updateSubEventsParser;
+    }
+
+    public function handle(ServerRequestInterface $request, string $eventId): ResponseInterface
+    {
+        $updates = $this->updateSubEventsParser->parse($request);
+
+        $updateSubEvents = [];
+        foreach ($updates as $update) {
+            $subEventUpdate = new SubEventUpdate($update['id']);
+
+            if (isset($update['status'])) {
+                $subEventUpdate = $subEventUpdate->withStatus(
+                    new Status(
+                        StatusType::fromNative($update['status']['type']),
+                        $this->parseReason($update)
+                    )
+                );
+            }
+
+            if (isset($update['bookingAvailability'])) {
+                $subEventUpdate = $subEventUpdate->withBookingAvailability(
+                    BookingAvailability::fromNative($update['bookingAvailability']['type'])
+                );
+            }
+
+            $updateSubEvents[] = $subEventUpdate;
+        }
+
+        $this->commandBus->dispatch(new UpdateSubEvents($eventId, ...$updateSubEvents));
```

Comment:
Nit: This will now throw if there are no `SubEventUpdate` objects inside the array. Would you suggest that I make the JSON schema for the validation require at least one subEvent so that error would get caught earlier and converted to the right ApiProblem (`body/invalid-data`), or do we just check here if the array count is `>= 1` and if not just return `NoContent` and do nothing, or remove the check from the command and also accept an empty array (which would make the command handler do nothing if I'm not mistaken so we'd also return `NoContent` without actually doing anything then).

Reply:
I did place the validation inside `UpdateSubEvents` if we would have another entry point then an API call, for example a CLI command, in that case the `UpdateSubEvents` would throw.
I also opted for a throw because in theory this requires at least one `SubEventUpdate` as defined by the type hinting, but how PHP handles the splat operator it also allows an empty array.

So I would keep it here, but also don't mind if it gets removed. It's more of a safety against a "missing" language feature of typed arrays or generics.

Reply:
I was just thinking that in theory it wouldn't harm if someone would dispatch this command without a any `SubEventUpdate` inside, it could just be a no-op then. But yeah I don't mind keeping it if we either add a check here then so we don't dispatch the command if there are no updates inside, OR we make the request validation expect at least one subEvent update and return an ApiProblem if you just do a request with literally `[]`

Reply:
Ok, sounds good. Then we keep it like it is for now. Easy enough to refactor something like this if it would become inconsistent.